### ISSUE TITLE
Running XESmartTarget Continuously for 24/7 Monitoring

### ISSUE THREAD

**Issue by 0x7FFFFFFFFFFFFFFF** [spaghettidba/XESmartTarget#31]
Hi,

I am new to using XESmartTarget and have gone through its documentation. From my understanding, in order to use XESmartTarget effectively, I need to run the executable continuously. Is that correct? If so, is there a built-in way to run it as a service? What is the recommended approach to ensure it runs 24/7 for monitoring a production SQL Server instance?

In the demos I've seen, the XESmartTarget executable is run in a command line. However, I am concerned that if the RDP session gets disconnected or logged off by someone, the program will exit, causing the monitoring to stop. I need a solution that ensures XESmartTarget runs continuously without being interrupted.

Any suggestions or best practices for achieving this would be greatly appreciated. Thank you in advance!

**Comment by spaghettidba** [comment]
Hi, XESmartTarget needs to run continuously, that's right. There is no built-in way to make XESmartTarget run as a service, but you can run it in a service wrapper like [NSSM](https://nssm.cc/). Basically, you register NSSM to run as a service (you can do so from the GUI, there's documentation on the website) and it runs whatever program you want at service start.
I don't recommend running XESmartTarget on the SQL Server box, use a different one if possible.
Hope this helps

**Comment by luke-mckee-ol** [comment]
NSSM was very useful for most of my set ups, however in some cases IT depts were not keen to install NSSM. An alternative to this we used was to have a powershell script that checked for a running process for XESmartTarget, we have a couple processes running so utilized [Import-Clixml](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/import-clixml?view=powershell-7.4) to track the processes. If the process wasn't running it kicked it off using [Start-Process](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/start-process?view=powershell-7.4) . We used the -PassThru argument for Start-Process to return the process object and [Export-Clixml](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/export-clixml?view=powershell-7.4) to persist the process information.

We then executed the script every 5 minutes in Windows Task Scheduler. It seems to work all good.

Hopefully this helps someone if they can't use NSSM :)
